src/format/reveal/format-reveal-theme.ts

```diff
@@ -85,7 +85,7 @@ export async function revealTheme(
   const revealUrl = pathWithForwardSlashes(revealDir);
   // escape to avoid pandoc markdown parsing from YAML default file
   // https://github.com/quarto-dev/quarto-cli/issues/9117
-  metadata[kRevealJsUrl] = pandocNativeStr(revealUrl);
+  metadata[kRevealJsUrl] = pandocNativeStr(revealUrl).mappedString().value;
```

Comment:
@cscheid is this the correct way to use the `pandocNativeStr` generator ? 

I looked at the test you made and fix my issue following it. 

I see the codegen function have `emit()` `mappedString()` and others. Probably a quick run through how all this works would help me. I see it is used for mermaid only for now.

Reply:
In this context, yes, that's exactly how I would use it. That codegen.ts module has features to maintain source information (useful for error messages) but those are not going to be usable in here, so yes, I think this looks good.

Reply:
> That codegen.ts module has features to maintain source information (useful for error messages)

Oh so that is what I am missing. I have seen Mapped string thing in other place, but not yet sure to know how it works; Thanks ! 

I'll add a note in changelog and merge when tests pases